### PR TITLE
[MRG] Adding Circleci 2.0

### LINKED ISSUE #61: ADD: Configure CircleCI
So that we can check the artifacts tab for builds of the docs

### PR DESCRIPTION
**Pull request by sappelhoff**
fixes #61 

The `Makefile` had a bunch of stuff from the MNE repo that didn't seem to apply for mne-bids. I removed that, please check and if in doubt, I can re-add it.

Some make commands also didn't work and I fixed them.

Let's see what CircleCI says to this.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+          echo "export PATH=~/miniconda/bin:$PATH" >> $BASH_ENV;
+    run:
+        command: |
+          conda update --yes --quiet conda;
+          conda env create -n testenv --yes pip python=3
+          source activate testenv;
+          conda install --yes --quiet numpy=1.13.1 scipy=0.19.1 matplotlib pandas=0.20.3
+          pip install flake8 pytest nose coverage coveralls
+          git clone --depth 1 https://github.com/mne-tools/mne-python.git -b maint/0.15
+          cd mne-python
```
Comment (agramfort) [inline .circleci/config.yml]: Cd mne_bids

Comment (sappelhoff) [reply .circleci/config.yml]: actually, that part is for installing mne-python. mne-bids is installed a line lower using `python setup.py develop`

Comment (agramfort) [reply .circleci/config.yml]: I would do it with

pip install -U https://api.github.com/repos/mne-tools/mne-python/zipball/maint/0.15

I should work.

Comment (jasmainak) [reply .circleci/config.yml]: The issue is that we use the data in `mne/tests/data` for our testing which isn't downloaded with `pip`. You need it there. But for CircleCI I think a normal pip install should do the job

**Comment by jasmainak** [comment]
Can you also add a badge in README.md so that we can see if CircleCI is failing on master or not?

**Comment by sappelhoff** [comment]
@jasmainak, CircleCI was not triggered.

The Travis error is not related to the code

**Comment by agramfort** [comment]
I see some builds but from master in : https://circleci.com/gh/mne-tools/mne-bids
can you try again? I just allowed build from forks.

**Comment by jasmainak** [comment]
Oh yes! works now. I couldn't find the setting for allowing builds from forks. @sappelhoff back to you :)

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                  conda create --yes --name testenv python=3
+                  source activate testenv;
+                  conda install --yes --quiet numpy scipy matplotlib pandas
+                  pip install flake8 pytest nose coverage coveralls
+                  git clone --depth 1 https://github.com/mne-tools/mne-python.git -b maint/0.15
+                  cd mne-python
+                  pip install --no-deps -e .
+                  cd ../
+                  python setup.py develop
+                  npm install -g https://github.com/INCF/bids-validator.git#0.26.0
```
Comment (jasmainak) [inline .circleci/config.yml]: why is this necessary?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                  ~/miniconda.sh -b -p ~/miniconda;
+                  echo "export PATH=~/miniconda/bin:$PATH" >> $BASH_ENV;
+            - run:
+                command: |
+                  conda update --yes --quiet conda;
+                  conda create --yes --name testenv python=3
+                  source activate testenv;
+                  conda install --yes --quiet numpy scipy matplotlib pandas
+                  pip install flake8 pytest nose coverage coveralls
+                  git clone --depth 1 https://github.com/mne-tools/mne-python.git -b maint/0.15
```
Comment (jasmainak) [inline .circleci/config.yml]: as @agramfort mentioned, let's do a regular pip install here

File: .circleci/config.yml (hunk 2 above)
Comment (jasmainak) [reply .circleci/config.yml]: I would use Circle CI only for building examples, not for tests ...

Comment (sappelhoff) [reply .circleci/config.yml]: okay, fair enough. Travis is enough, I'll skip the testing :-)

**Comment by sappelhoff** [comment]
mmhh ... why does building the doc fail? Sphinx is installed ...

**Comment by jasmainak** [comment]
@sappelhoff you might want to leave a few `echo` commands in the `yml` to check if `sphinx` is indeed installed. Otherwise, you can ssh into the system to check what's the problem

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            - run:
+                name: Setup Python environment
+                command: |
+                  conda update --yes --quiet conda;
+                  conda env create -f environment.yml
+                  source activate mne_bids;
+                  python setup.py develop
+            - run:
+                name: Build the documentation
+                command: |
```
Comment (agramfort) [inline .circleci/config.yml]: maybe try to insert:

source activate mne_bids

here too.

**Comment by jasmainak** [comment]
Squashed and merged. Thanks @sappelhoff !
